pkg/envfuncs/kind_funcs.go

```diff
@@ -85,7 +85,7 @@ func CreateKindClusterWithConfig(clusterName, image, configFilePath string) env.
 		}
 
 		// store entire cluster value in ctx for future access using the cluster name
-		return context.WithValue(ctx, kindContextKey(clusterName), k), nil
+		return context.WithValue(ctx, KindContextKey(clusterName), k), nil
```

Comment:
Ok I understand better what you are trying to achieve by making `kindContextKey` visible. My comment here is that the `envfuncs` package is designed to expose callable functions (i.e. `envfuncs.CreateKindCluster(...)`, etc). Exposing KindContextKey as a type may be confusing (a little, not much).

Proposal:
* Revert `KindContextKey` to private scope.
* Change `clusterNames` (TestMain) to `[]string`
* Add new `envfuncs` to retrieve cluster from context as shown below:

```go
func TestSomething(...) {
Assess(fmt.Sprintf("Deployment is running successfully - %s", clusterNames[0]), func(ctx context.Context, t *testing.T, config *envconf.Config) context.Context {
			cluster, ok := envfuncs.GetKindClusterWithCtx(ctx, clusterNames[0]))
			checkPodStatus(t, cluster.GetKubeconfig(), clusterNames[0])
			return ctx
		})
...
}
```

Or something similar as above.  That way you can avoid all of the `string` converstions and keep things private.

Reply:
Makes sense. Let me get that changed accordingly.